kata-design-requirements.md

```diff
@@ -0,0 +1,82 @@
+## Mandatory requirements
+
+The Kata Containers runtime **MUST** fulfill all requirements below:
+
+### OCI compatibility
+The Kata Containers runtime **MUST** implement the OCI runtime specification and support all the OCI runtime operations.
+
+### `runc` CLI compatibility
+In theory, being OCI compatible should be enough. In practice the a Kata Containers runtime should comply the latest *stable* `runc` CLI. In particular, it **MUST** implement with the following `runc` commands:
+
+* `create`
+* `delete`
+* `exec`
+* `kill`
+* `list`
+* `pause`
+* `ps`
+* `state`
+* `start`
+* `version`
+
+and the following command line options:
+* `--console-socket`
+* `--pid-file`
+
+### CRI and Kubernetes support
+The Kata Containers project **MUST** provide 2 interfaces for CRI shims to be able to manage hardware virtualization based Kubernetes pods and containers:
+-  An OCI and `runc` compatible command line interface, as decribed in the previous section. This interface is used by e.g. the `CRI-O` and `cri-containerd` CRI implementations
+- A hardware virtualization runtime library API for CRI shims to consume and provide a more CRI native implementation. The `frakti` CRI shim is one example of such consumer.
+
+### Multiple hardware architectures support
+The Kata Containers runtime **MUST NOT** be architecture specific. It should be able to support multiple hardware architectures and provide a pluggable and flexible design for adding support for additional ones.
+
+### Multiple hypervisor support
+The Kata Containers runtime **MUST NOT** be tied to any specific hardware virtualization technology, hypervisor or virtual machine monitor implementation.
+It should support multiple hypervisors and provide a pluggable and flexible design for adding support for additional ones.
+
+#### Nesting
+The Kata Containers runtime **MUST** support nested virtualization environments.
+
+### Networking
+
+* The Kata Containers runtime **MUST** be able to support any CNI plugin.
+* The Kata Containers runtime **MUST** be able to support both legacy and IPv6 networks.
+
+### I/O
+
+#### Devices direct assignment
+In order for containers to directly consume host hardware resources, the Kata Containers runtime **MUST** provide containers with secure pass through to generic devices like e.g. GPUs, SRIOV, RDMA, QAT, by leveraging I/O virtualization technologies (IOMMU, interrupt remapping, etc...).
+
+#### Acceleration
+The Kata Containers runtime **MUST** support accelerated and user space based I/O operations for networking (e.g. DPDK) and storage through `vhost-user` sockets.
+
+#### Scalability
+The Kata Containers runtime **MUST** support scalable I/O through the SRIOV technology.
+
+
+### Virtualization overhead reduction
+One of the compelling aspects of containers is its minimal overhead compared to bare metal applications. A container runtime should strive for keeping that overhead to a minimum in order to provide the expected user experience.
```

Comment:
Suggested rewrite: A compelling aspect of containers is the minimal overhead compared to bare metal applications. A container runtime should keep the overhead to a minimum in order to provide the expected user experience.

Reply:
agreed